### PR TITLE
Add RV32GC support to macaw-riscv

### PR DESCRIPTION
**Pull request by bboston7**
This change adds support for RV32GC RISC-V binaries.  Specifically, it:

* Updates the return matcher to recognize returns in 32-bit binaries
* Updates detection of unsupported binaries to allow RV32GC binaries
* Adds RV32GC versions of the RV64GC tests

### CODE REVIEW

File: macaw-riscv/src/Data/Macaw/RISCV/Identify.hs [hunk 1]
```diff
@@ -62,7 +62,12 @@ matchReturn :: RISCVConstraints rv
             -> Maybe (Some (MA.AbsValue w))
 matchReturn rvRepr absProcState ip = G.withRV rvRepr $ do
   MC.BVAnd _ addr (MC.BVValue _ mask) <- MC.valueAsApp ip
-  guard (mask == 0xfffffffffffffffe)
+  let maskVal = case G.rvBaseArch rvRepr of
+                  G.RV64Repr -> 0xfffffffffffffffe
+                  G.RV32Repr -> 0xfffffffe
+                  _ -> error ( "Unsupported base architecture: "
+                            ++ (show (G.rvBaseArch rvRepr)))
```
Comment (travitch) [inline macaw-riscv/src/Data/Macaw/RISCV/Identify.hs:69]: What do you think about converting the mask into a `BV w` from bv-sized and then making `maskVal = sext w (mkBV (knownNat @8) 0xfe)`

That would be resilient to any architecture width.

File: macaw-riscv/src/Data/Macaw/RISCV/Identify.hs [hunk 2]
```diff
@@ -62,7 +62,12 @@ matchReturn :: RISCVConstraints rv
             -> Maybe (Some (MA.AbsValue w))
```
Comment (travitch) [inline macaw-riscv/src/Data/Macaw/RISCV/Identify.hs:62]: Do we get to fix the FIXME comment on this function with this change?

**Review by travitch — APPROVED** [review]

File: macaw-riscv/src/Data/Macaw/RISCV/Identify.hs (hunk 1 above)
Comment (bboston7) [reply macaw-riscv/src/Data/Macaw/RISCV/Identify.hs]: Done!

File: macaw-riscv/src/Data/Macaw/RISCV/Identify.hs (hunk 2 above)
Comment (bboston7) [reply macaw-riscv/src/Data/Macaw/RISCV/Identify.hs:67]: Yep!  Good catch.  I removed the FIXME.

**Pull request merged**